newIDE/app/src/EventsSheet/ParameterFields/VariableField.js

```diff
@@ -118,3 +118,12 @@ export default class VariableField extends Component<Props, State> {
     );
   }
 }
+
+export const VariableWithIcon = (value, iconPath, tooltip = '') => {
+  return (
+    <span title={tooltip}>
+      <img style={{maxWidth:48}} width={'auto'} height={24} src={iconPath} />
```

Comment:
You're sure about your style?
width={24} and height={24} should work?

Be sure to use only the bare minimum styling that is useful. Also add some right margin :)

Reply:
The thing is that it will either squish or stretch them horizontally if it is a set value. Thats why I am using maxWidth as well. Will this be ok to have If I import the styling from one place?

Reply:
We still need to find a better styling because all icons should be a fixed square size. It's super weird to have some objects take more width or height than other, it's creating a confusing result.
Make sure that all images fit in a fixed size box, without stretching. I don't have the solution right now but it's entirely possible. It's what is done for thumbnail in the objects list. Do the same but with a size of 20, 24, or 16, according to what is better (and that should be the same size as other icons)

Reply:
I find that object thumbnails look much better as 24 in height, but if they are affecting other things we have to keep them at 16. In any case I think that we shouldn't treat them the same as icons. Its very likely that most thumbnails will be portrait in aspect ratio. Their meaning is also a bit more important - as they visually communicate which object is affected. That is probably why they are slightly bigger in construct2 than icons

Reply:
@4ian the old var icons are wider than they are tall. I forced them to a square constraint as you requested, but that makes them a bit blurry. I think that a lot of icons will need to be redesigned to have a 1x1 ratio